src/Features/Core/Portable/RemoveUnusedMembers/AbstractAvoidUnusedMembersCodeFixProvider.cs

```diff
@@ -0,0 +1,116 @@
+        protected abstract void AdjustDeclarators(HashSet<TFieldDeclarationSyntax> fieldDeclarators, HashSet<SyntaxNode> declarators);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics[0], c)),
+                context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document,
+            ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor,
+            CancellationToken cancellationToken)
+        {
+            return FixWithEditorAsync(document, editor, diagnostics, cancellationToken);
+        }
+
+        private async Task FixWithEditorAsync(
+            Document document, SyntaxEditor editor, ImmutableArray<Diagnostic> diagnostics,
+            CancellationToken cancellationToken)
+        {
+            var declarators = new HashSet<SyntaxNode>();
+            var fieldDeclarators = new HashSet<TFieldDeclarationSyntax>();
+
+            // Compute declarators to remove, and also track common field declarators.
+            foreach (var diagnostic in diagnostics)
+            {
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var diagnosticSpan = diagnostic.Location.SourceSpan;
+                var symbolName = diagnostic.Properties[AvoidUnusedMembersDiagnosticAnalyzer.UnunsedMemberNameProperty];
+                var symbolKind = diagnostic.Properties[AvoidUnusedMembersDiagnosticAnalyzer.UnunsedMemberKindProperty];
+
+                var node = GetTopmostSyntaxNodeForSymbolDeclaration(root.FindNode(diagnosticSpan),
+                    isSymbolDeclarationNode: n => n != null && semanticModel.GetDeclaredSymbol(n, cancellationToken)?.Name == symbolName);
+
+                declarators.Add(node);
+                if (symbolKind == nameof(SymbolKind.Field))
+                {
+                    var fieldDeclarator = node.FirstAncestorOrSelf<TFieldDeclarationSyntax>();
+                    fieldDeclarators.Add(fieldDeclarator);
+                }
+            }
+
+            if (fieldDeclarators.Count > 0)
+            {
+                AdjustDeclarators(fieldDeclarators, declarators);
+            }
+
+            foreach (var declarator in declarators)
+            {
+                editor.RemoveNode(declarator);
+            }
+        }
+
+        protected virtual SyntaxNode GetTopmostSyntaxNodeForSymbolDeclaration(SyntaxNode syntaxNode, Func<SyntaxNode, bool> isSymbolDeclarationNode)
+        {
+            return syntaxNode.FirstAncestorOrSelf(isSymbolDeclarationNode);
+        }
```

Comment:
this helper function seems unnecessary.  you could just inline directly into the callsite.

Reply:
It is virtual, overridden in VB.

---
In reply to: [212729134](https://github.com/dotnet/roslyn/pull/29511#discussion_r212729134) [](ancestors = 212729134)

Reply:
See above:

> in general, we don't use that pattern. Instead, we keep things abstract, making C# and VB have to decide what behavior they want. It's weird to have a defualt behavior that is then only used by one language, and then have the other language override it. In essence, this is saying "VB is overriding C#'s behavior". And instead we want to say: it's up to each language to decide what the correct behavior is.

> Also see later note: we have a service that already does this work. So i don't think this extension point is needed in this feature. Look up ISymbolDeclarationService :)

#unbydesigned :)